tests/regex/RegexTest.php

```diff
@@ -308,8 +308,11 @@ public function testReplace(
       ->toEqual($expected);
   }
 
-  public static function provideReplaceWith(): varray<mixed> {
-    return varray[
+  /*HHAST_IGNORE_ERROR[DataProviderTypes]
+    `(function(nothing): string)` has fewer tokens than `(function(Regex\Match): string)`
+    and the linter will therefore not be able to compare them.*/
```

Comment:
This looks fine for now, but in the future, would it be too hard to update the linter to work with AST nodes instead of tokens, to avoid issues like this? (allowing `mixed` instead of `(function...)` might require that change too, since the number of tokens in `(function...)` is unpredictable)